pkg/registry/apis/iam/register.go

```diff
@@ -346,6 +410,33 @@ func (b *IdentityAccessManagementAPIBuilder) mutateUser(_ context.Context, a adm
 	return nil
 }
 
+func (b *IdentityAccessManagementAPIBuilder) mutateServiceAccount(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error {
+	saObj, ok := a.GetObject().(*iamv0.ServiceAccount)
+	if !ok {
+		return nil
+	}
+
+	ns, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return err
+	}
+
+	// External service accounts have None org role by default
+	if saObj.Spec.External {
+		saObj.Spec.Role = iamv0.ServiceAccountOrgRoleNone
+	}
+
+	prefix := serviceaccounts.ServiceAccountPrefix
+	if saObj.Spec.External {
+		prefix = serviceaccounts.ExtSvcLoginPrefix(ns.OrgID)
+	}
+	saObj.Spec.Login = strings.ToLower(serviceaccounts.GenerateLogin(prefix, ns.OrgID, saObj.Spec.Title))
```

Comment:
ExtSvcLoginPrefix already embeds the org ID: fmt.Sprintf("%s%d-%s", ServiceAccountPrefix, orgID, ExtSvcPrefix)

GenerateLogin again appends “-<orgID>-<name>”, so external logins become “sa<orgID>-extsvc-<orgID>-<name>” (org ID duplicated, and note missing hyphen after “sa”).

This doesn't seem to be inline with the current generation

Reply:
which normally generates [sa-1-extsvc-grafana-irm-app](https://ops.grafana-ops.net/org/serviceaccounts/dee559j3pz9j4a)

Reply:
Thanks, it's been fixed: https://github.com/grafana/grafana/pull/110411/files#diff-842069985d64b6b7eeb28a29600fa30fb7b7a33963363dabe0d7bb558e44bb63R92